packages/telemetry/src/context-aware-slog.js

```diff
@@ -0,0 +1,475 @@
+ *    'crank.deliveryNum': Slog['deliveryNum'];
+ *    'crank.num': Slog['crankNum'];
+ *    'crank.type': Slog['crankType'];
+ *    'crank.vatID': Slog['vatID'];
+ *    init: boolean;
+ *    replay: boolean;
+ *    'run.id': string;
+ *    'run.num': string;
+ *    'run.trigger.blockHeight': Slog['blockHeight'];
+ *    'run.trigger.msgIdx': number;
+ *    'run.trigger.sender': Slog['sender'];
+ *    'run.trigger.source': Slog['source'];
+ *    'run.trigger.time': Slog['blockTime'];
+ *    'run.trigger.txHash': string;
+ *    'run.trigger.type': string;
+ *  }>
+ * } Context
+ *
+ * @typedef {{
+ *  'chain-id': string;
+ *  'crank.syscallNum'?: Slog['syscallNum'];
+ *  'process.uptime': Slog['monotime'];
+ *  timestamp: Slog['time'];
+ * } & Context} LogAttributes
+ */
+
+const DATABASE_FILE_PATH =
+  process.env.SLOG_CONTEXT_DATABASE_FILE_PATH || '/state/slog-db.sqlite3';
+const DATABASE_TABLE_NAME = 'context';
+const DATABASE_TRIGGER_CONTEXT_KEY_NAME = 'trigger-context';
+
+const DATABASE_TYPES = {
+  INTEGER: 'INTEGER',
+  TEXT: 'TEXT',
+};
+
+const DATABASE_SCHEMA = {
+  key: `${DATABASE_TYPES.TEXT} PRIMARY KEY`,
+  'run.id': DATABASE_TYPES.TEXT,
+  'run.num': DATABASE_TYPES.TEXT,
+  'run.trigger.blockHeight': DATABASE_TYPES.INTEGER,
+  'run.trigger.msgIdx': DATABASE_TYPES.INTEGER,
+  'run.trigger.sender': DATABASE_TYPES.TEXT,
+  'run.trigger.source': DATABASE_TYPES.TEXT,
+  'run.trigger.time': DATABASE_TYPES.INTEGER,
+  'run.trigger.txHash': DATABASE_TYPES.TEXT,
+  'run.trigger.type': DATABASE_TYPES.TEXT,
+};
+
+const SLOG_TYPES = {
+  CLIST: 'clist',
+  CONSOLE: 'console',
+  CRANK: {
+    RESULT: 'crank-result',
+    START: 'crank-start',
+  },
+  DELIVER: 'deliver',
+  DELIVER_RESULT: 'deliver-result',
+  KERNEL: {
+    INIT: {
+      FINISH: 'kernel-init-finish',
+      START: 'kernel-init-start',
+    },
+  },
+  REPLAY: {
+    FINISH: 'finish-replay',
+    START: 'start-replay',
+  },
+  // eslint-disable-next-line no-restricted-syntax
+  RUN: {
+    FINISH: 'cosmic-swingset-run-finish',
+    START: 'cosmic-swingset-run-start',
+  },
+  SWINGSET: {
+    AFTER_COMMIT_STATS: 'cosmic-swingset-after-commit-stats',
+    BEGIN_BLOCK: 'cosmic-swingset-begin-block',
+    BOOTSTRAP_BLOCK: {
+      FINISH: 'cosmic-swingset-bootstrap-block-finish',
+      START: 'cosmic-swingset-bootstrap-block-start',
+    },
+    BRIDGE_INBOUND: 'cosmic-swingset-bridge-inbound',
+    COMMIT: {
+      FINISH: 'cosmic-swingset-commit-finish',
+      START: 'cosmic-swingset-commit-start',
+    },
+    DELIVER_INBOUND: 'cosmic-swingset-deliver-inbound',
+    END_BLOCK: {
+      FINISH: 'cosmic-swingset-end-block-finish',
+      START: 'cosmic-swingset-end-block-start',
+    },
+  },
+  SYSCALL: 'syscall',
+  SYSCALL_RESULT: 'syscall-result',
+};
+
+const getDatabaseInstance = async () => {
+  if (!existsSync(DATABASE_FILE_PATH))
+    closeSync(openSync(DATABASE_FILE_PATH, 'w'));
+
+  const databaseInstance = sqlite(DATABASE_FILE_PATH, { fileMustExist: true });
```

Comment:
Currently there is no need for a DB. I added DB in case we need to add more context or things in future. If we don't see this happening right now, we can move to a simple json file.